sha3/src/macros.rs

```diff
@@ -424,10 +424,10 @@ macro_rules! impl_cshake {
                 );
                 buffer.digest_blocks(customization, |blocks| {
                     for block in blocks {
-                        state.absorb_block(block);
+                        state.absorb_block(&block);
```

Comment:
Are you sure that you need borrow here? The closure receives `&[Block<Self>]`, so `block` should be `&Block`.